src/operator/src/bulk_insert.rs

```diff
@@ -40,32 +41,19 @@ impl Inserter {
     pub async fn handle_bulk_insert(
         &self,
         table: TableRef,
-        decoder: &mut FlightDecoder,
-        data: FlightData,
+        raw_flight_data: FlightData,
+        record_batch: RecordBatch,
+        schema_bytes: Bytes,
     ) -> error::Result<AffectedRows> {
         let table_info = table.table_info();
         let table_id = table_info.table_id();
         let db_name = table_info.get_db_string();
-        let decode_timer = metrics::HANDLE_BULK_INSERT_ELAPSED
-            .with_label_values(&["decode_request"])
-            .start_timer();
-        let body_size = data.data_body.len();
-        // Build region server requests
-        let message = decoder
-            .try_decode(&data)
-            .context(error::DecodeFlightDataSnafu)?
-            .context(error::NotSupportedSnafu {
-                feat: "bulk insert RecordBatch with dictionary arrays",
-            })?;
-        let FlightMessage::RecordBatch(record_batch) = message else {
-            return Ok(0);
-        };
-        decode_timer.observe_duration();
 
         if record_batch.num_rows() == 0 {
             return Ok(0);
         }
 
+        let body_size = raw_flight_data.data_body.len();
         // TODO(yingwen): Fill record batch impure default values.
```

Comment:
Maybe we have to leave a comment that if we fill default values here, then we can't reuse the `raw_flight_data`.
```suggestion
        // TODO(yingwen): Fill record batch impure default values. Note that we should override `raw_flight_data` if we have to fill defaults.
```